src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -4746,23 +4744,23 @@ LogicalResult ONNXTopKOp::verify() {
           [](const Value &op) { return !hasShapeAndRank(op); }))
     return success(); // Won't be able to do any checking at this stage.
 
+  // K's rank must be zero or one.
```

Comment:
I think we still cannot come to this point if X is unranked because of `llvm::any_of`. I suggest to check K's rank and X's rank separately when needed instead of `llvm::any_of`.

Reply:
Ok, modified this member function implementation as you suggest. Please take another look.